compiler/expression/block_compiler.rs

```diff
@@ -138,7 +138,7 @@ fn resolve_type_for_variable<'a, Snapshot: ReadableSnapshot>(
             .iter()
             .map(|type_| match type_ {
                 Type::Attribute(attribute_type) => attribute_type
-                    .get_value_type(context.snapshot, context.type_manager)
+                    .get_value_type_without_source(context.snapshot, context.type_manager)
```

Comment:
I was torn apart between 
`get_value_type() -> ValueType` + `get_value_type_with_source -> (ValueType, AttributeType)`
and
`get_value_type() -> (ValueType, AttributeType)` + `get_value_type_without_source -> ValueType`

For now, I've chosen the second option, because all the other non-`_defined` interfaces return both the value and the source. Plus, we duplicate the `type_manager`'s call (`get_value_type()` -> `type_manager.get_value_type()`), and `without_source` is just an additional method. Still, not sure if it's pleasant to use.